doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,91 @@
+---
+author: Kaiyu Wang KaiyuWang16/kawa@microsoft.com
+created on: 2019-10-16
+last updated: 2019-10-16
+issue id: #605
+---
+
+# Search in Terminal
+
+## Abstract
+
+This spec is for feature request #605 "Search". It goes over the details of a new feature that allows users to search text in Terminal, within one tab or from all tabs. Expected behavior and design of this feature is included. Besides, future possible follow-up works are also addressed. 
+
+## Inspiration
+
+One of the superior features of iTerm2 is it's content search. The search comes in two variants: search from active tab and search from all tabs. In almost any editor, there is an roughly equivalent string search. We also want to realize search experience in Terminal. There will be two variants, search within one tab or from multiple tabs. We will start with one-tab search implementation. 
+
+## Solution Design
+
+Our ultimate goal is to provide both search within one tab and search from all tabs experiences. But we can start with one-tab search. The search experience should have following features:
+
+1. The search is triggered by keybindings "Ctrl + F". This coincides with other editors. In the future we will also consider adding a "Find" in the dropdown menu. 

```

Comment:
I'd probably avoid calling out a specific keybinding here. Instead, I'd say that we'll add a `ShortcutAction` that can be used to toggle the Find dialog, bound to <kbd>ctrl+F</kbd> by default.

I'd also refrain from mentioning the "find in the dropdown menu" feature here - You discuss it in greater detail in the Further Considerations section, let's just contain the discussion to that section.

Reply:
changed